clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -384,9 +418,12 @@
                                                                         0),
                                                               new Field("responses",
                                                                       new ArrayOf(FETCH_RESPONSE_TOPIC_V0)));
+    // Fetch response V2 is the same as fetch response V1. The version number is bumped up to indicate that the Message
```

Comment:
Right, perhaps we can make this clearer in the comment. Sth like the following:

Even though fetch response v2 has the same protocol as  v1, the record set in the response is different. In v1, record set only includes messages of v0 (magic byte 0). In v2, record set can include messages of v0 and v1 (magic byte 0 and 1). For details, see ref{ByteBufferMessageSet}.